### PR TITLE
Only group delete visible items

### LINKED ISSUE #4745: Selected but not visible items in a list will get removed
In any filterable item-list-layout, if an item is selected (checked) and then filtered out by a search, when the remove icon (-) is activated all selected items (visible or not) are offered for deletion. This is probably not a good idea.

https://user-images.githubusercontent.com/40840436/150882594-651cb8d0-f4cd-469d-805f-33af92baa082.mov

We could either make the filtered list available to the delete function, or automatically uncheck any items not visible. The latter is probably easier but a worse UX (user might expect the non-visible item to be selected when it is made visible again).

Version: 5.3.4 (this is a long-standing issue)

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

fixes https://github.com/lensapp/lens/issues/4745

https://user-images.githubusercontent.com/8225332/152186942-30b2829a-d52a-47db-b0c3-5b33693f9c5f.mov

### CODE REVIEW

**Review by aleksfront — DISMISSED** [review]
LGTM

**Review by aleksfront — DISMISSED** [review]

File: src/renderer/components/+helm-releases/releases.tsx [hunk 1]
```diff
@@ -134,10 +134,12 @@ class NonInjectedHelmReleases extends Component<Dependencies & Props> {
...
-      removeSelectedItems() {
-        return Promise.all(
-          releases.get().filter((release) => release.isSelected).map((release) => release.delete()),
-        );
+      pickOnlySelected: (releases) => {
+        return releases.filter(release => release.isSelected);
+      },
+
+      removeItems: async (releases) => {
+        await Promise.all(releases.map(release => release.delete()));
```
Comment (jim-docker) [inline src/renderer/components/+helm-releases/releases.tsx:142]: Port forwards may need a similar fix? I just removed all of mine thinking I was removing just one

Comment (Nokel81) [reply src/renderer/components/+helm-releases/releases.tsx:142]: Ah thanks for the catch

**Review by ixrock — DISMISSED** [review]

**Comment by Nokel81** [comment]
@ixrock @jim-docker @aleksfront PTALA, I have fixed the PortForwardStore

**Review by ixrock — APPROVED** [review]

**Review by jim-docker — APPROVED** [review]

**Pull request merged**